### DISCUSSION TITLE
Difference between @storybook/jest and @storybook/addon-jest?

### DISCUSSION THREAD

**Discussion by cmun2** [storybookjs/storybook#23894] [Help]
I'm currently working on `Vue 3` Project and willing to apply `Jest + VTU(Vue Test Utils) + Storybook 7`.

I want to try using `Jest + VTU` for the unit tests in logic and `Storybook 7` for the UI testing.

There are choices in `Storybook 7` for `Vue 3` such as downloading `@storybook/jest` or `@storybook/addon-jest` in `main.js` file.

What I discovered in the documents so far is that `@storybook/jest` has to be linked up with interactions add-on which I already got installed as `@storybook/addon-interactions` and `@storybook/addon-jest` can be used as what it is.
 
What are the specific differences between them in usage? 
Additionally, I already set my `Vue` project with `Jest` downloaded with Vue Default Setup. Is `@storybook/jest` or `@storybook/addon-jest` working separately with this options which I have to remove `jest.config.js` file or these are linking with the current `Jest` config file?

**Comment by kasir-barati** [comment]
Having same questions @cmun2, did you manage to find the answers to these questions?

Also I was wondering on how to tell `@storybook/jest` to use a different attribute for `getByTestId`, speaking of configuring jest, it is super easy when we are dealing with `jest`, you can create a `jest.setup.ts` and pass it to the `jest.config.ts` as `setupFilesAfterEnv` but in Storybook configured by Nx, I do not have any clue about how to do it.

https://github.com/kasir-barati/you-say/blob/main/apps/frontend/jest.config.ts

**Reply by kasir-barati** [reply]
So after digging and digging for about 30 minutes or so I ended up with these two really useful and good enough resources to solve my* issue:
1. https://github.com/storybookjs/storybook/issues/23930#issuecomment-1729931715
2. https://storybook.js.org/blog/storybook-test/

*I said **my** since only needed to change the `data-testid` to `data-test` and that was good enough for me.

I hope this helps you. If you are still unsure take a look at my repo, I've implemented it there. 

BTW **I appreciate it if you could also give it a star**: [kasir-barati/you-say](https://github.com/kasir-barati/you-say)
